lib/phoenix/controller.ex

```diff
@@ -1389,7 +1389,7 @@ defmodule Phoenix.Controller do
         cur_uri  = MyAppWeb.Endpoint.struct_url()
         cur_path = Phoenix.Controller.current_path(conn, params)
 
-        MyAppWeb.Router.Helpers.url(%URI{cur_uri | scheme: "https}) <> cur_path
```

Comment:
Sorry, fixed just quotation mark.
By the way, I'm trying this code and it does not seem to work
is `%Struct{variable | struct_field: value}` valid code?

Reply:
@sysashi the syntax is explained in the Elixir documentation https://elixir-lang.org/getting-started/keywords-and-maps.html#maps

> Maps have the following syntax for updating a key’s value:
> 
> ```elixir
> iex> map = %{:a => 1, 2 => :b}
> %{2 => :b, :a => 1}
> 
> iex> %{map | 2 => "two"}
> %{2 => "two", :a => 1}
> iex> %{map | :c => 3}
> ** (KeyError) key :c not found in: %{2 => :b, :a => 1}
> ```
> The syntax above requires the given key to exist. It cannot be used to add new keys. For example, using it with the :c key failed because there is no :c in the map.

Reply:
Yeah, that's what I've used in first commit. However `%URI{cur_uri | scheme: "https"}` is not this case. Maybe I'm missing something here, have not been practicing elixir for some time :(

Reply:
Oh, you're right, sorry for confusion.